src/towncrier/test/test_build.py

```diff
@@ -143,6 +143,32 @@ def test_in_different_dir_config_option(self, runner):
         self.assertEqual(0, result.exit_code)
         self.assertTrue((project_dir / "NEWS.rst").exists())
 
+    @with_isolated_runner
+    def test_in_different_dir_with_nondefault_newsfragments_directory(self, runner):
+        """
+        Config location differs from the base directory for news file and fragments.
+
+        This is useful when multiple projects share one towncrier configuration.
+        The default `newsfragments` setting already supports this scenario so here
+        we test that custom settings also do.
```

Comment:
I think is best to just focus on the core functionality that is tested here.

```suggestion
        Using the `--dir` CLI argument, the NEWS file can
        be generated in a sub-directory from fragments
        that are relatives to that sub-directory.
        
        The path passed to `--dir` becomes the
        working directory.
```